numpy/lib/format.py

```diff
@@ -1011,3 +1011,15 @@ def isfileobj(f):
         return True
     except OSError:
         return False
+
+
+class NumpyUnpickler(pickle.Unpickler):
+    """
+    A thin wrapper for :py:class:`pickle.Unpickler` that
+    allows to load legacy array pickles with ``dtype=object``.
```

Comment:
```suggestion
    allows to load legacy array pickles.
```

It's not just `dtype=object` that's problematic for generic pickle files.